xbmc/addons/AddonManager.cpp

```diff
@@ -215,21 +218,38 @@ VECADDONS CAddonMgr::GetAvailableUpdates() const
   VECADDONS installed;
   VECADDONS all_addons;
   m_database.GetRepositoryContent(all_addons);
-  std::map<std::string, AddonPtr> last_versions;
+  std::map<std::string, AddonPtr> latest_private_versions;
+  std::map<std::string, AddonPtr> latest_trusted_versions;
   for (const auto& addon : all_addons)
   {
-    const auto& latest_known = last_versions.find(addon->ID());
-    if (latest_known == last_versions.end() || addon->Version() > latest_known->second->Version())
-      last_versions[addon->ID()] = std::move(addon);
+    if (IsFromTrustedRepo(addon))
+    {
+      AddAddonIfLatest(addon, latest_trusted_versions);
+    }
+    else
+    {
+      AddAddonIfLatest(addon, latest_private_versions);
+    }
   }
 
+  CLog::Log(LOGDEBUG, "ADDONS: trusted versions count : {}", latest_trusted_versions.size());
+  CLog::Log(LOGDEBUG, "ADDONS: private versions count : {}", latest_private_versions.size());
+
   GetAddonsForUpdate(installed);
+
   for (const auto& addon : installed)
   {
-    const auto& remote = last_versions.find(addon->ID());
-    if (remote != last_versions.end() && remote->second->Version() > addon->Version())
-      updates.emplace_back(std::move(remote->second));
+    CLog::Log(LOGDEBUG, "ADDONS: update check: addonID = {} / Origin = {}",
+        addon->ID(), addon->Origin());
```

Comment:
see https://github.com/xbmc/xbmc/pull/18111#issuecomment-653470189

Reply:
What is this addon ID: b6a50484-93a0-4afb-a01c-8d17e059feda. Should it not just be repository.xbmc.org?

Reply:
this list is generated from the installed addons on the system, where we're checking updates.
i think this some kind of "system-origin":
https://github.com/xbmc/xbmc/blob/9897aa7a1deff9911d43d7ae0649c16750396f3c/xbmc/addons/Addon.h#L23
not sure how to deal with this right now

Reply:
I think you are right.

Reply:
Maybe ORIGIN_SYSTEM should also be a trusted repo?

Reply:
was my idea too, but i think the origin_system addons are checked somewhere else. as we have only split the single map into two maps. i‘ll count previous last_version map now... should also have 456...

Reply:
for the record:
`2020-07-03 15:06:25.080 T:2940     INFO <general>: ADDONS: last_versions count : 456`
counted without this pullrequest.. i think we're good at that point

Reply:
👍